memo/managers.py

```diff
@@ -73,3 +74,63 @@ def get_card_amount_in_deck(self, deck):
 class DeckManager(models.Manager):
     def get_decks_by_owner(self, user):
         return self.filter(owner=user)
+
+
+class CardAnswerHistoryManager(models.Manager):
+    def get_consecutive_days(self, user):
+        from .models import Card
+
+        max_daily_date_times = (
+            self.filter(owner=user)
+            .extra(select={"day": "date( created_at )"})
+            .values("day")
+            .annotate(latest_datetime=Max("created_at"))
+        )
+
+        objects = (
+            self.filter(
+                created_at__in=[
+                    entry["latest_datetime"] for entry in max_daily_date_times
+                ]
+            )
+            .values("card", "created_at")
+            .order_by("-created_at")
+        )
+
+        result = []
+
+        if objects:
+            answered_today = False
+            today = datetime.today()
+            count = 0
+            one_day_less = 1
+
+            for index in range(0, len(objects)):
```

Comment:
This is a code smell in python. If you want to iterate over a list and wants to the index as well, you can use the `enumerate` function. It returns both the index and the object. Ex:

```python
a_list = ["Lelis", "Lucia"]
for index, item in enumerate(a_list):
    print(index, item)

# 1, Lelis
# 2, Lucia
```
```suggestion
            for index, object in enumerate(objects):
```

Reply:
Okay, thank you for showing this function. :+1: